network/p2p/p2p.go

```diff
@@ -228,11 +229,17 @@ func (n *p2pNetwork) Close() error {
 func (n *p2pNetwork) getConnector() (chan peer.AddrInfo, error) {
 	connector := make(chan peer.AddrInfo, connectorQueueSize)
 	go func() {
-		// Wait for own subnets to be subscribed to and updated.
-		// TODO: wait more intelligently with a channel.
-		time.Sleep(8 * time.Second)
 		ctx, cancel := context.WithCancel(n.ctx)
 		defer cancel()
+
+		// Wait for own subnets to be subscribed to and updated.
+		// TODO: wait more intelligently with a channel.
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(8 * time.Second):
```

Comment:
magic number, i would move that number (+TODO) to the top of the package + add some docs

Reply:
Same as https://github.com/ssvlabs/ssv/pull/2375#discussion_r2185523032, resolving.

Please reopen if you have any objections